src/object/cli_shard.c

```diff
@@ -528,6 +561,9 @@ dc_obj_shard_rw(struct dc_obj_shard *shard, enum obj_rpc_opc opc,
 	rw_args.shard_args = args;
 	/* remember the sgl to copyout the data inline for fetch */
 	rw_args.rwaa_sgls = (opc == DAOS_OBJ_RPC_FETCH) ? sgls : NULL;
+	rw_args.maps = args->api_args->maps;
+	if (rw_args.maps != NULL)
+		orw->orw_flags |= ORF_CREATE_MAP;
```

Comment:
we may define ORF_IOM_ALL and ORF_IOM_LOHI to differentiate that user may only want to know the lo/hi (need not transfer all recx in that case).
but can be added later (I can make the change in my patch later)

Reply:
yes indeed. in array_get_size we do not care for all recx in this case.